src/eigen/preconditioners.jl

```diff
@@ -33,7 +33,8 @@ mutable struct PreconditionerTPA{T <: Real}
 end
 
 function PreconditionerTPA(basis::PlaneWaveBasis{T}, kpt::Kpoint; default_shift=1) where T
-    kin = Vector{T}([sum(abs2, G + kpt.coordinate_cart)
+    scaling = only([t for t in basis.model.term_types if t isa Kinetic]).scaling_factor
+    kin = Vector{T}([scaling * sum(abs2, G + kpt.coordinate_cart)
```

Comment:
If you are doing that you might as well re-use the `kinetic_energies` of the `TermKinetic`.

Reply:
yes but we don't know the index of kpt in basis. That's a bit of a problem with our API for the kpoint datastructure. Hard to fix it without a backref from kpoint to the basis, which you didn't want

Reply:
Well you could look that up ... but arguably that's no cleaner.

Reply:
Yeah, I'm wondering now why we don't have a field in kpoint with the basis and the index of this kpoint in the basis?

Reply:
Because circular pointers are bad. They kick you in the back eventually. I'm fine with storing the index, though.